python/cudf/cudf/core/dataframe.py

```diff
@@ -1475,7 +1475,18 @@ def reindex(
 
         if idx is not None:
             idx = idx if isinstance(idx, Index) else as_index(idx)
-            if df.index.dtype != idx.dtype:
+
+            from cudf.core import MultiIndex
+
+            is_multi = type(idx) == MultiIndex
+            if is_multi:
+                idx_match = [
+                    [x for x in df.index._source_data.dtypes] for i in idx
+                ] == [[x for x in idx._source_data.dtypes] for i in idx]
+            else:
+                idx_match = [df.index.dtype == idx.dtype]
```

Comment:
Can you explain what `idx_match` is used for? I'm not quite following the behavior here.